ibc-core/ics23-commitment/types/src/error.rs

```diff
@@ -13,6 +13,10 @@ pub enum CommitmentError {
     EmptyMerkleRoot,
     /// empty verified value
     EmptyVerifiedValue,
+    /// empty proof specs
+    EmptyProofSpecs,
+    /// empty depth range: [{0}, {1}]
+    EmptyDepthRange(i32, i32),
```

Comment:
Just a point caught my eye, we shouldn’t allow negative values too. I opened up another issue (https://github.com/cosmos/ibc-rs/issues/1108) for that. Let’s use `u32` here.

Reply:
Let's use `i32` here as this is the error report. I check for non-negative values in `.validate()`

67415960